### PR TITLE
[25292] TUD API Route Authentication

### PR DESCRIPTION
**Pull request by bramleyjl**
## Description of change
While steps have been taken to place the TUD frontend behind SOCKS and add basic auth, the TUD API route is available publicly; direct requests to it would allow anyone to receive all of the test user information available in the app, including user passwords. This change will add authentication based on a shared secret key passed through a JWT.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#25292

## Things to know about this PR
This PR is a WIP.

### CODE REVIEW

File: modules/test_user_dashboard/app/controllers/test_user_dashboard/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
 module TestUserDashboard
   class ApplicationController < ActionController::API
+    include SentryLogging
+    before_action :require_jwt
+    # SECRET = 'password123'
+
+    def require_jwt
+      # payload from example
+      exp = Time.now.to_i + 300
+      payload = { 'iss': 'tud.staging.va.gov',
```
Comment (AGarcia-Clarity) [inline modules/test_user_dashboard/app/controllers/test_user_dashboard/application_controller.rb:14]: tud.vfs.va.gov

**Comment by AGarcia-Clarity** [comment]
I know we previously went over this but I can't exactly remember.

1. Are we using symmetric or asymmetric keys?
2. Can we add other routes on the module?
> - `/login` generates the JWT token
> - `/refresh` refresh JWT token based on expiry date
> - `/invalidate` invalidates the JWT token

We can sync on this later if we need to.

**Comment by bramleyjl** [comment]
> I know we previously went over this but I can't exactly remember.
> 
> 1. Are we using symmetric or asymmetric keys?
> 2. Can we add other routes on the module?
> 
> > * `/login` generates the JWT token
> > * `/refresh` refresh JWT token based on expiry date
> > * `/invalidate` invalidates the JWT token
> 
> We can sync on this later if we need to.

1. Per @joeniquette's suggestion we're going to be using asymmetrical keys, the example implementation in this PR has been updated with this.
2. Are these routes going to be necessary for the implementation? I was under the impression that JWT management is supposed to be happening within the TUD frontend application and the backend is just attempting to decrypt the token passed to it to make sure that the caller is the TUD frontend and not another system/user.

**Review by joeniquette — APPROVED** [review]
LGTM

**Comment by ericboehs** [comment]
@bramleyjl Can this PR be closed?

**Comment by ericboehs** [comment]
Errr, I closed it. Please reopen if you still need this.

**Review by ericboehs — APPROVED** [review]

**Pull request merged**